src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -40,6 +41,7 @@
 tsql_dialect.sets("reserved_keywords").clear()
 tsql_dialect.sets("unreserved_keywords").clear()
 tsql_dialect.sets("reserved_keywords").update(RESERVED_KEYWORDS)
+tsql_dialect.sets("reserved_keywords").difference_update(BUILTIN_FUNCTIONS)
```

Comment:
```suggestion
#tsql_dialect.sets("reserved_keywords").difference_update(BUILTIN_FUNCTIONS)
# I think this may fix an issue.
tsql_dialect.sets("reserved_keywords").update(BUILTIN_FUNCTIONS)
```